### DISCUSSION TITLE
Calling k_thread_create with K_FOREVER timeout and never starting it

### DISCUSSION THREAD

**Discussion by adrdub** [zephyrproject-rtos/zephyr#52017] [Q&A] [answered]
Hello,

I couldn't find the answer in Zephyr's docs for following aspect: is it safe to call `k_thread_create` function with `K_FOREVER` timeout multiple times and never starting created threads? I see no API to for eg. cancel such 'non-started' thread, however its ID is already known. Or maybe `k_thread_abort` should be called? What'd be recommended for such use case?

BR,
Adrian

**Comment by yashi** [comment, marked as answer]
I don't know what is the definition of _safe_ for you.  But your system should work without those threads, given that each `k_thread_create()` is given a unique set of arguments.

To remove a thread from the Kernel, you should call `k_thread_abort()`, then free all resources it had.  Just grep through the source code and you'll see many usage of `k_thread_abort()`.

**Reply by adrdub** [reply]
That was my guess, however I didn't find clear statement in docs for this specific use case. Thanks!
